### DISCUSSION TITLE
Default as null

### DISCUSSION THREAD

**Discussion by zbeyens** [47ng/nuqs#490] [Ideas] [answered]
For a better DX, I want to use a default value instead of `null` to unset a query. Here is the util I've came with:

```ts
import { useQueryState } from 'nuqs';

export const useNullableQueryState = <T>(
  nullValue: T,
  ...args: Parameters<typeof useQueryState<T>>
) => {
  const [state, setState] = useQueryState(...args);

  return [
    (state ?? nullValue) as T,
    (value: T) => {
      void setState(value === nullValue ? null : (value as any));
    },
  ] as const;
};
```

Is there a `nuqs` way to achieve that? If not, it would be a great addition.

**Comment by franky47** [comment]
Would #488 fit your purpose?

It's been discussed a few times now, I'll probably add an option to clear the query when setting to the default (with a disclaimer in the docs about changing defaults).

**Reply by franky47** [reply, marked as answer]
@zbeyens I've opened #493 to implement this option, let me know what you think there.
